src/librustc_mir/interpret/operand.rs

```diff
@@ -598,10 +598,18 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 throw_inval!(TooGeneric),
             ty::ConstKind::Unevaluated(def_id, substs) => {
                 let instance = self.resolve(def_id, substs)?;
-                return Ok(OpTy::from(self.const_eval_raw(GlobalId {
+                let param_env = if self.tcx.is_static(def_id) {
+                    ty::ParamEnv::reveal_all()
+                } else {
+                    self.param_env
+                };
```

Comment:
This looks subtle, please add a comment explaining what is going on.

Reply:
Also Cc @eddyb who keeps telling me our `ParamEnv` are all wrong. I know nothing about how to pick the right now.^^